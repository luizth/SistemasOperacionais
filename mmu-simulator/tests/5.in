4
3 99
2 6
3 8
0 25
3 70
2 118
1 37
1 120
0 25
2 64
2 95
0 80
3 48
3 122
3 113
1 81
1 123
0 67
3 54
0 67
1 73
0 3
0 112
0 14
1 81
1 63
2 33
2 43
1 13
1 92
1 68
2 6
3 22
2 41
1 51
3 100
2 41
3 74
1 94
2 4
1 102
0 71
2 62
2 88
1 120
1 18
1 50
1 73
1 19
0 90
3 60
2 117
2 72
2 47
0 49
0 26
3 70
0 20
2 69
3 8
0 87
3 32
3 9
3 106
1 1
1 37
0 82
2 110
2 77
1 94
2 95
0 82
2 36
2 91
1 81
0 5
1 66
3 57
1 50
0 121
2 31
2 21
2 95
2 95
1 40
1 81
0 101
2 95
3 93
1 11
2 103
2 96
2 72
3 78
0 5
0 46
3 60
1 37
2 39
2 103
0 24
0 89
1 7
1 75
3 10
2 69
1 107
0 124
2 53
1 7
2 33
3 28
1 73
2 36
0 104
2 96
3 22
3 27
1 94
2 2
1 37
1 75
1 55
2 36
2 31
2 61
1 13
1 102
3 32
2 2
3 22
3 59
2 83
2 105
1 35
3 76
3 115
0 0
2 117
0 49
1 68
1 111
3 86
1 66
3 28
2 33
3 127
0 114
1 84
1 68
0 116
3 79
1 94
2 17
0 89
1 84
3 22
1 68
0 0
0 116
2 105
3 113
3 115
0 5
3 8
1 51
2 61
1 51
2 72
2 77
3 59
1 35
0 16
3 70
2 62
1 66
2 23
1 125
3 106
0 87
0 121
1 81
3 106
3 48
3 32
0 97
2 91
0 71
3 28
3 126
0 98
2 62
0 71
2 69
2 117
2 21
2 117
2 23
2 56
1 111
2 6
0 30
3 126
3 8
3 113
1 109
3 8
0 90
2 77
1 94
1 123
2 39
2 83
3 65
3 74
2 103
3 113
0 108
0 44
3 8
