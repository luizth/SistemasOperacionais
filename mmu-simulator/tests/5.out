Process 3 accessed page 99 - frame 0
Process 2 accessed page 6 - frame 1
Process 3 accessed page 8 - frame 2
Process 0 accessed page 25 - frame 3
Process 3 accessed page 70 - frame 4
Process 2 accessed page 118 - frame 5
Process 1 accessed page 37 - frame 6
Process 1 accessed page 120 - frame 7
Page 25 in memory (frame 3)!
Process 0 accessed page 25 - frame 3
Memory is full! Frame 0 was replaced (page 99 - pid 3)
Process 2 accessed page 64 - frame 0
Memory is full! Frame 1 was replaced (page 6 - pid 2)
Process 2 accessed page 95 - frame 1
Memory is full! Frame 2 was replaced (page 8 - pid 3)
Process 0 accessed page 80 - frame 2
Memory is full! Frame 3 was replaced (page 25 - pid 0)
Process 3 accessed page 48 - frame 3
Memory is full! Frame 4 was replaced (page 70 - pid 3)
Process 3 accessed page 122 - frame 4
Memory is full! Frame 5 was replaced (page 118 - pid 2)
Process 3 accessed page 113 - frame 5
Memory is full! Frame 6 was replaced (page 37 - pid 1)
Process 1 accessed page 81 - frame 6
Memory is full! Frame 7 was replaced (page 120 - pid 1)
Process 1 accessed page 123 - frame 7
Memory is full! Frame 0 was replaced (page 64 - pid 2)
Process 0 accessed page 67 - frame 0
Memory is full! Frame 1 was replaced (page 95 - pid 2)
Process 3 accessed page 54 - frame 1
Page 67 in memory (frame 0)!
Process 0 accessed page 67 - frame 0
Memory is full! Frame 2 was replaced (page 80 - pid 0)
Process 1 accessed page 73 - frame 2
Memory is full! Frame 3 was replaced (page 48 - pid 3)
Process 0 accessed page 3 - frame 3
Memory is full! Frame 4 was replaced (page 122 - pid 3)
Process 0 accessed page 112 - frame 4
Memory is full! Frame 5 was replaced (page 113 - pid 3)
Process 0 accessed page 14 - frame 5
Page 81 in memory (frame 6)!
Process 1 accessed page 81 - frame 6
Memory is full! Frame 6 was replaced (page 81 - pid 1)
Process 1 accessed page 63 - frame 6
Memory is full! Frame 7 was replaced (page 123 - pid 1)
Process 2 accessed page 33 - frame 7
Memory is full! Frame 0 was replaced (page 67 - pid 0)
Process 2 accessed page 43 - frame 0
Memory is full! Frame 1 was replaced (page 54 - pid 3)
Process 1 accessed page 13 - frame 1
Memory is full! Frame 2 was replaced (page 73 - pid 1)
Process 1 accessed page 92 - frame 2
Memory is full! Frame 3 was replaced (page 3 - pid 0)
Process 1 accessed page 68 - frame 3
Memory is full! Frame 4 was replaced (page 112 - pid 0)
Process 2 accessed page 6 - frame 4
Memory is full! Frame 5 was replaced (page 14 - pid 0)
Process 3 accessed page 22 - frame 5
Memory is full! Frame 6 was replaced (page 63 - pid 1)
Process 2 accessed page 41 - frame 6
Memory is full! Frame 7 was replaced (page 33 - pid 2)
Process 1 accessed page 51 - frame 7
Memory is full! Frame 0 was replaced (page 43 - pid 2)
Process 3 accessed page 100 - frame 0
Page 41 in memory (frame 6)!
Process 2 accessed page 41 - frame 6
Memory is full! Frame 1 was replaced (page 13 - pid 1)
Process 3 accessed page 74 - frame 1
Memory is full! Frame 2 was replaced (page 92 - pid 1)
Process 1 accessed page 94 - frame 2
Memory is full! Frame 3 was replaced (page 68 - pid 1)
Process 2 accessed page 4 - frame 3
Memory is full! Frame 4 was replaced (page 6 - pid 2)
Process 1 accessed page 102 - frame 4
Memory is full! Frame 5 was replaced (page 22 - pid 3)
Process 0 accessed page 71 - frame 5
Memory is full! Frame 6 was replaced (page 41 - pid 2)
Process 2 accessed page 62 - frame 6
Memory is full! Frame 7 was replaced (page 51 - pid 1)
Process 2 accessed page 88 - frame 7
Memory is full! Frame 0 was replaced (page 100 - pid 3)
Process 1 accessed page 120 - frame 0
Memory is full! Frame 1 was replaced (page 74 - pid 3)
Process 1 accessed page 18 - frame 1
Memory is full! Frame 2 was replaced (page 94 - pid 1)
Process 1 accessed page 50 - frame 2
Memory is full! Frame 3 was replaced (page 4 - pid 2)
Process 1 accessed page 73 - frame 3
Memory is full! Frame 4 was replaced (page 102 - pid 1)
Process 1 accessed page 19 - frame 4
Memory is full! Frame 5 was replaced (page 71 - pid 0)
Process 0 accessed page 90 - frame 5
Memory is full! Frame 6 was replaced (page 62 - pid 2)
Process 3 accessed page 60 - frame 6
Memory is full! Frame 7 was replaced (page 88 - pid 2)
Process 2 accessed page 117 - frame 7
Memory is full! Frame 0 was replaced (page 120 - pid 1)
Process 2 accessed page 72 - frame 0
Memory is full! Frame 1 was replaced (page 18 - pid 1)
Process 2 accessed page 47 - frame 1
Memory is full! Frame 2 was replaced (page 50 - pid 1)
Process 0 accessed page 49 - frame 2
Memory is full! Frame 3 was replaced (page 73 - pid 1)
Process 0 accessed page 26 - frame 3
Memory is full! Frame 4 was replaced (page 19 - pid 1)
Process 3 accessed page 70 - frame 4
Memory is full! Frame 5 was replaced (page 90 - pid 0)
Process 0 accessed page 20 - frame 5
Memory is full! Frame 6 was replaced (page 60 - pid 3)
Process 2 accessed page 69 - frame 6
Memory is full! Frame 7 was replaced (page 117 - pid 2)
Process 3 accessed page 8 - frame 7
Memory is full! Frame 0 was replaced (page 72 - pid 2)
Process 0 accessed page 87 - frame 0
Memory is full! Frame 1 was replaced (page 47 - pid 2)
Process 3 accessed page 32 - frame 1
Memory is full! Frame 2 was replaced (page 49 - pid 0)
Process 3 accessed page 9 - frame 2
Memory is full! Frame 3 was replaced (page 26 - pid 0)
Process 3 accessed page 106 - frame 3
Memory is full! Frame 4 was replaced (page 70 - pid 3)
Process 1 accessed page 1 - frame 4
Memory is full! Frame 5 was replaced (page 20 - pid 0)
Process 1 accessed page 37 - frame 5
Memory is full! Frame 6 was replaced (page 69 - pid 2)
Process 0 accessed page 82 - frame 6
Memory is full! Frame 7 was replaced (page 8 - pid 3)
Process 2 accessed page 110 - frame 7
Memory is full! Frame 0 was replaced (page 87 - pid 0)
Process 2 accessed page 77 - frame 0
Memory is full! Frame 1 was replaced (page 32 - pid 3)
Process 1 accessed page 94 - frame 1
Memory is full! Frame 2 was replaced (page 9 - pid 3)
Process 2 accessed page 95 - frame 2
Page 82 in memory (frame 6)!
Process 0 accessed page 82 - frame 6
Memory is full! Frame 3 was replaced (page 106 - pid 3)
Process 2 accessed page 36 - frame 3
Memory is full! Frame 4 was replaced (page 1 - pid 1)
Process 2 accessed page 91 - frame 4
Memory is full! Frame 5 was replaced (page 37 - pid 1)
Process 1 accessed page 81 - frame 5
Memory is full! Frame 6 was replaced (page 82 - pid 0)
Process 0 accessed page 5 - frame 6
Memory is full! Frame 7 was replaced (page 110 - pid 2)
Process 1 accessed page 66 - frame 7
Memory is full! Frame 0 was replaced (page 77 - pid 2)
Process 3 accessed page 57 - frame 0
Memory is full! Frame 1 was replaced (page 94 - pid 1)
Process 1 accessed page 50 - frame 1
Memory is full! Frame 2 was replaced (page 95 - pid 2)
Process 0 accessed page 121 - frame 2
Memory is full! Frame 3 was replaced (page 36 - pid 2)
Process 2 accessed page 31 - frame 3
Memory is full! Frame 4 was replaced (page 91 - pid 2)
Process 2 accessed page 21 - frame 4
Memory is full! Frame 5 was replaced (page 81 - pid 1)
Process 2 accessed page 95 - frame 5
Page 95 in memory (frame 5)!
Process 2 accessed page 95 - frame 5
Memory is full! Frame 6 was replaced (page 5 - pid 0)
Process 1 accessed page 40 - frame 6
Memory is full! Frame 7 was replaced (page 66 - pid 1)
Process 1 accessed page 81 - frame 7
Memory is full! Frame 0 was replaced (page 57 - pid 3)
Process 0 accessed page 101 - frame 0
Page 95 in memory (frame 5)!
Process 2 accessed page 95 - frame 5
Memory is full! Frame 1 was replaced (page 50 - pid 1)
Process 3 accessed page 93 - frame 1
Memory is full! Frame 2 was replaced (page 121 - pid 0)
Process 1 accessed page 11 - frame 2
Memory is full! Frame 3 was replaced (page 31 - pid 2)
Process 2 accessed page 103 - frame 3
Memory is full! Frame 4 was replaced (page 21 - pid 2)
Process 2 accessed page 96 - frame 4
Memory is full! Frame 5 was replaced (page 95 - pid 2)
Process 2 accessed page 72 - frame 5
Memory is full! Frame 6 was replaced (page 40 - pid 1)
Process 3 accessed page 78 - frame 6
Memory is full! Frame 7 was replaced (page 81 - pid 1)
Process 0 accessed page 5 - frame 7
Memory is full! Frame 0 was replaced (page 101 - pid 0)
Process 0 accessed page 46 - frame 0
Memory is full! Frame 1 was replaced (page 93 - pid 3)
Process 3 accessed page 60 - frame 1
Memory is full! Frame 2 was replaced (page 11 - pid 1)
Process 1 accessed page 37 - frame 2
Memory is full! Frame 3 was replaced (page 103 - pid 2)
Process 2 accessed page 39 - frame 3
Memory is full! Frame 4 was replaced (page 96 - pid 2)
Process 2 accessed page 103 - frame 4
Memory is full! Frame 5 was replaced (page 72 - pid 2)
Process 0 accessed page 24 - frame 5
Memory is full! Frame 6 was replaced (page 78 - pid 3)
Process 0 accessed page 89 - frame 6
Memory is full! Frame 7 was replaced (page 5 - pid 0)
Process 1 accessed page 7 - frame 7
Memory is full! Frame 0 was replaced (page 46 - pid 0)
Process 1 accessed page 75 - frame 0
Memory is full! Frame 1 was replaced (page 60 - pid 3)
Process 3 accessed page 10 - frame 1
Memory is full! Frame 2 was replaced (page 37 - pid 1)
Process 2 accessed page 69 - frame 2
Memory is full! Frame 3 was replaced (page 39 - pid 2)
Process 1 accessed page 107 - frame 3
Memory is full! Frame 4 was replaced (page 103 - pid 2)
Process 0 accessed page 124 - frame 4
Memory is full! Frame 5 was replaced (page 24 - pid 0)
Process 2 accessed page 53 - frame 5
Page 7 in memory (frame 7)!
Process 1 accessed page 7 - frame 7
Memory is full! Frame 6 was replaced (page 89 - pid 0)
Process 2 accessed page 33 - frame 6
Memory is full! Frame 7 was replaced (page 7 - pid 1)
Process 3 accessed page 28 - frame 7
Memory is full! Frame 0 was replaced (page 75 - pid 1)
Process 1 accessed page 73 - frame 0
Memory is full! Frame 1 was replaced (page 10 - pid 3)
Process 2 accessed page 36 - frame 1
Memory is full! Frame 2 was replaced (page 69 - pid 2)
Process 0 accessed page 104 - frame 2
Memory is full! Frame 3 was replaced (page 107 - pid 1)
Process 2 accessed page 96 - frame 3
Memory is full! Frame 4 was replaced (page 124 - pid 0)
Process 3 accessed page 22 - frame 4
Memory is full! Frame 5 was replaced (page 53 - pid 2)
Process 3 accessed page 27 - frame 5
Memory is full! Frame 6 was replaced (page 33 - pid 2)
Process 1 accessed page 94 - frame 6
Memory is full! Frame 7 was replaced (page 28 - pid 3)
Process 2 accessed page 2 - frame 7
Memory is full! Frame 0 was replaced (page 73 - pid 1)
Process 1 accessed page 37 - frame 0
Memory is full! Frame 1 was replaced (page 36 - pid 2)
Process 1 accessed page 75 - frame 1
Memory is full! Frame 2 was replaced (page 104 - pid 0)
Process 1 accessed page 55 - frame 2
Memory is full! Frame 3 was replaced (page 96 - pid 2)
Process 2 accessed page 36 - frame 3
Memory is full! Frame 4 was replaced (page 22 - pid 3)
Process 2 accessed page 31 - frame 4
Memory is full! Frame 5 was replaced (page 27 - pid 3)
Process 2 accessed page 61 - frame 5
Memory is full! Frame 6 was replaced (page 94 - pid 1)
Process 1 accessed page 13 - frame 6
Memory is full! Frame 7 was replaced (page 2 - pid 2)
Process 1 accessed page 102 - frame 7
Memory is full! Frame 0 was replaced (page 37 - pid 1)
Process 3 accessed page 32 - frame 0
Memory is full! Frame 1 was replaced (page 75 - pid 1)
Process 2 accessed page 2 - frame 1
Memory is full! Frame 2 was replaced (page 55 - pid 1)
Process 3 accessed page 22 - frame 2
Memory is full! Frame 3 was replaced (page 36 - pid 2)
Process 3 accessed page 59 - frame 3
Memory is full! Frame 4 was replaced (page 31 - pid 2)
Process 2 accessed page 83 - frame 4
Memory is full! Frame 5 was replaced (page 61 - pid 2)
Process 2 accessed page 105 - frame 5
Memory is full! Frame 6 was replaced (page 13 - pid 1)
Process 1 accessed page 35 - frame 6
Memory is full! Frame 7 was replaced (page 102 - pid 1)
Process 3 accessed page 76 - frame 7
Memory is full! Frame 0 was replaced (page 32 - pid 3)
Process 3 accessed page 115 - frame 0
Memory is full! Frame 1 was replaced (page 2 - pid 2)
Process 0 accessed page 0 - frame 1
Memory is full! Frame 2 was replaced (page 22 - pid 3)
Process 2 accessed page 117 - frame 2
Memory is full! Frame 3 was replaced (page 59 - pid 3)
Process 0 accessed page 49 - frame 3
Memory is full! Frame 4 was replaced (page 83 - pid 2)
Process 1 accessed page 68 - frame 4
Memory is full! Frame 5 was replaced (page 105 - pid 2)
Process 1 accessed page 111 - frame 5
Memory is full! Frame 6 was replaced (page 35 - pid 1)
Process 3 accessed page 86 - frame 6
Memory is full! Frame 7 was replaced (page 76 - pid 3)
Process 1 accessed page 66 - frame 7
Memory is full! Frame 0 was replaced (page 115 - pid 3)
Process 3 accessed page 28 - frame 0
Memory is full! Frame 1 was replaced (page 0 - pid 0)
Process 2 accessed page 33 - frame 1
Memory is full! Frame 2 was replaced (page 117 - pid 2)
Process 3 accessed page 127 - frame 2
Memory is full! Frame 3 was replaced (page 49 - pid 0)
Process 0 accessed page 114 - frame 3
Memory is full! Frame 4 was replaced (page 68 - pid 1)
Process 1 accessed page 84 - frame 4
Memory is full! Frame 5 was replaced (page 111 - pid 1)
Process 1 accessed page 68 - frame 5
Memory is full! Frame 6 was replaced (page 86 - pid 3)
Process 0 accessed page 116 - frame 6
Memory is full! Frame 7 was replaced (page 66 - pid 1)
Process 3 accessed page 79 - frame 7
Memory is full! Frame 0 was replaced (page 28 - pid 3)
Process 1 accessed page 94 - frame 0
Memory is full! Frame 1 was replaced (page 33 - pid 2)
Process 2 accessed page 17 - frame 1
Memory is full! Frame 2 was replaced (page 127 - pid 3)
Process 0 accessed page 89 - frame 2
Page 84 in memory (frame 4)!
Process 1 accessed page 84 - frame 4
Memory is full! Frame 3 was replaced (page 114 - pid 0)
Process 3 accessed page 22 - frame 3
Page 68 in memory (frame 5)!
Process 1 accessed page 68 - frame 5
Memory is full! Frame 4 was replaced (page 84 - pid 1)
Process 0 accessed page 0 - frame 4
Page 116 in memory (frame 6)!
Process 0 accessed page 116 - frame 6
Memory is full! Frame 5 was replaced (page 68 - pid 1)
Process 2 accessed page 105 - frame 5
Memory is full! Frame 6 was replaced (page 116 - pid 0)
Process 3 accessed page 113 - frame 6
Memory is full! Frame 7 was replaced (page 79 - pid 3)
Process 3 accessed page 115 - frame 7
Memory is full! Frame 0 was replaced (page 94 - pid 1)
Process 0 accessed page 5 - frame 0
Memory is full! Frame 1 was replaced (page 17 - pid 2)
Process 3 accessed page 8 - frame 1
Memory is full! Frame 2 was replaced (page 89 - pid 0)
Process 1 accessed page 51 - frame 2
Memory is full! Frame 3 was replaced (page 22 - pid 3)
Process 2 accessed page 61 - frame 3
Page 51 in memory (frame 2)!
Process 1 accessed page 51 - frame 2
Memory is full! Frame 4 was replaced (page 0 - pid 0)
Process 2 accessed page 72 - frame 4
Memory is full! Frame 5 was replaced (page 105 - pid 2)
Process 2 accessed page 77 - frame 5
Memory is full! Frame 6 was replaced (page 113 - pid 3)
Process 3 accessed page 59 - frame 6
Memory is full! Frame 7 was replaced (page 115 - pid 3)
Process 1 accessed page 35 - frame 7
Memory is full! Frame 0 was replaced (page 5 - pid 0)
Process 0 accessed page 16 - frame 0
Memory is full! Frame 1 was replaced (page 8 - pid 3)
Process 3 accessed page 70 - frame 1
Memory is full! Frame 2 was replaced (page 51 - pid 1)
Process 2 accessed page 62 - frame 2
Memory is full! Frame 3 was replaced (page 61 - pid 2)
Process 1 accessed page 66 - frame 3
Memory is full! Frame 4 was replaced (page 72 - pid 2)
Process 2 accessed page 23 - frame 4
Memory is full! Frame 5 was replaced (page 77 - pid 2)
Process 1 accessed page 125 - frame 5
Memory is full! Frame 6 was replaced (page 59 - pid 3)
Process 3 accessed page 106 - frame 6
Memory is full! Frame 7 was replaced (page 35 - pid 1)
Process 0 accessed page 87 - frame 7
Memory is full! Frame 0 was replaced (page 16 - pid 0)
Process 0 accessed page 121 - frame 0
Memory is full! Frame 1 was replaced (page 70 - pid 3)
Process 1 accessed page 81 - frame 1
Page 106 in memory (frame 6)!
Process 3 accessed page 106 - frame 6
Memory is full! Frame 2 was replaced (page 62 - pid 2)
Process 3 accessed page 48 - frame 2
Memory is full! Frame 3 was replaced (page 66 - pid 1)
Process 3 accessed page 32 - frame 3
Memory is full! Frame 4 was replaced (page 23 - pid 2)
Process 0 accessed page 97 - frame 4
Memory is full! Frame 5 was replaced (page 125 - pid 1)
Process 2 accessed page 91 - frame 5
Memory is full! Frame 6 was replaced (page 106 - pid 3)
Process 0 accessed page 71 - frame 6
Memory is full! Frame 7 was replaced (page 87 - pid 0)
Process 3 accessed page 28 - frame 7
Memory is full! Frame 0 was replaced (page 121 - pid 0)
Process 3 accessed page 126 - frame 0
Memory is full! Frame 1 was replaced (page 81 - pid 1)
Process 0 accessed page 98 - frame 1
Memory is full! Frame 2 was replaced (page 48 - pid 3)
Process 2 accessed page 62 - frame 2
Page 71 in memory (frame 6)!
Process 0 accessed page 71 - frame 6
Memory is full! Frame 3 was replaced (page 32 - pid 3)
Process 2 accessed page 69 - frame 3
Memory is full! Frame 4 was replaced (page 97 - pid 0)
Process 2 accessed page 117 - frame 4
Memory is full! Frame 5 was replaced (page 91 - pid 2)
Process 2 accessed page 21 - frame 5
Page 117 in memory (frame 4)!
Process 2 accessed page 117 - frame 4
Memory is full! Frame 6 was replaced (page 71 - pid 0)
Process 2 accessed page 23 - frame 6
Memory is full! Frame 7 was replaced (page 28 - pid 3)
Process 2 accessed page 56 - frame 7
Memory is full! Frame 0 was replaced (page 126 - pid 3)
Process 1 accessed page 111 - frame 0
Memory is full! Frame 1 was replaced (page 98 - pid 0)
Process 2 accessed page 6 - frame 1
Memory is full! Frame 2 was replaced (page 62 - pid 2)
Process 0 accessed page 30 - frame 2
Memory is full! Frame 3 was replaced (page 69 - pid 2)
Process 3 accessed page 126 - frame 3
Memory is full! Frame 4 was replaced (page 117 - pid 2)
Process 3 accessed page 8 - frame 4
Memory is full! Frame 5 was replaced (page 21 - pid 2)
Process 3 accessed page 113 - frame 5
Memory is full! Frame 6 was replaced (page 23 - pid 2)
Process 1 accessed page 109 - frame 6
Page 8 in memory (frame 4)!
Process 3 accessed page 8 - frame 4
Memory is full! Frame 7 was replaced (page 56 - pid 2)
Process 0 accessed page 90 - frame 7
Memory is full! Frame 0 was replaced (page 111 - pid 1)
Process 2 accessed page 77 - frame 0
Memory is full! Frame 1 was replaced (page 6 - pid 2)
Process 1 accessed page 94 - frame 1
Memory is full! Frame 2 was replaced (page 30 - pid 0)
Process 1 accessed page 123 - frame 2
Memory is full! Frame 3 was replaced (page 126 - pid 3)
Process 2 accessed page 39 - frame 3
Memory is full! Frame 4 was replaced (page 8 - pid 3)
Process 2 accessed page 83 - frame 4
Memory is full! Frame 5 was replaced (page 113 - pid 3)
Process 3 accessed page 65 - frame 5
Memory is full! Frame 6 was replaced (page 109 - pid 1)
Process 3 accessed page 74 - frame 6
Memory is full! Frame 7 was replaced (page 90 - pid 0)
Process 2 accessed page 103 - frame 7
Memory is full! Frame 0 was replaced (page 77 - pid 2)
Process 3 accessed page 113 - frame 0
Memory is full! Frame 1 was replaced (page 94 - pid 1)
Process 0 accessed page 108 - frame 1
Memory is full! Frame 2 was replaced (page 123 - pid 1)
Process 0 accessed page 44 - frame 2
Memory is full! Frame 3 was replaced (page 39 - pid 2)
Process 3 accessed page 8 - frame 3
